sys/net/gnrc/link_layer/gnrc_lorawan/gnrc_lorawan_mlme.c

```diff
@@ -0,0 +1,354 @@
+            if (!gnrc_lorawan_mac_acquire(mac)) {
+                mlme_confirm->status = -EBUSY;
+                return;
+            }
+
+            if (mac->mlme.backoff_budget < 0) {
+                mlme_confirm->status = -EDQUOT;
+                break;
+            }
+            memcpy(mac->appskey, mlme_request->param.join.appkey, LORAMAC_APPKEY_LEN);
+            mlme_confirm->status = gnrc_lorawan_send_join_request(mac, mlme_request->param.join.deveui,
+                                                                  mlme_request->param.join.appeui, mlme_request->param.join.appkey, mlme_request->param.join.dr);
+            break;
+        case MLME_LINK_CHECK:
+            mac->mlme.pending_mlme_opts |= GNRC_LORAWAN_MLME_OPTS_LINK_CHECK_REQ;
+            mlme_confirm->status = GNRC_LORAWAN_REQ_STATUS_DEFERRED;
+            break;
+        case MLME_SET:
+            _mlme_set(mac, mlme_request, mlme_confirm);
+            break;
+        case MLME_GET:
+            _mlme_get(mac, mlme_request, mlme_confirm);
+            break;
+        case MLME_RESET:
+            gnrc_lorawan_reset(mac);
+            mlme_confirm->status = GNRC_LORAWAN_REQ_STATUS_SUCCESS;
+            break;
+        default:
+            break;
+    }
+}
+
+int _fopts_mlme_link_check_req(gnrc_lorawan_t *mac, lorawan_buffer_t *buf)
+{
+    (void) mac;
+    if (buf) {
+        assert(buf->index + GNRC_LORAWAN_CID_SIZE <= buf->size);
+        buf->data[buf->index++] = GNRC_LORAWAN_CID_LINK_CHECK_REQ_ANS;
+    }
+
+    return GNRC_LORAWAN_CID_SIZE;
+}
+
+static int _mlme_link_check_ans(gnrc_lorawan_t *mac, lorawan_buffer_t *fopt)
+{
+    if (fopt->index + GNRC_LORAWAN_FOPT_LINK_ANS_SIZE > fopt->size) {
+        return -EINVAL;
+    }
+    fopt->index++;
+
+    mlme_confirm_t *mlme_confirm = _mlme_allocate(mac);
+    mlme_confirm->param.link_req.margin = fopt->data[fopt->index++];
+    mlme_confirm->param.link_req.num_gateways = fopt->data[fopt->index++];
+
+    mlme_confirm->type = MLME_LINK_CHECK;
+    mlme_confirm->status = GNRC_LORAWAN_REQ_STATUS_SUCCESS;
+    mac->netdev.event_callback(&mac->netdev, NETDEV_EVENT_MLME_CONFIRM);
+
+    mac->mlme.pending_mlme_opts &= ~GNRC_LORAWAN_MLME_OPTS_LINK_CHECK_REQ;
+
+    return 0;
+}
+
+void gnrc_lorawan_process_fopts(gnrc_lorawan_t *mac, uint8_t *fopts, size_t size)
+{
+    if (!fopts || !size) {
+        goto out;
+    }
+
+    lorawan_buffer_t buf;
+    _buffer_reset(&buf, fopts, size);
+
+    while (buf.index < buf.size) {
+        switch (*(buf.data)) {
+            case GNRC_LORAWAN_CID_LINK_CHECK_REQ_ANS:
+                if (_mlme_link_check_ans(mac, &buf) < 0) {
+                    goto out;
+                }
+                break;
+            default:
+                goto out;
+        }
+    }
+out:
+    return;
+}
+
+uint8_t gnrc_lorawan_build_options(gnrc_lorawan_t *mac, lorawan_buffer_t *buf)
+{
+    size_t size = 0;
+
+    size += (mac->mlme.pending_mlme_opts & GNRC_LORAWAN_MLME_OPTS_LINK_CHECK_REQ) ?
```

Comment:
Why not simplify to:

```
return (mac->mlme.pending_mlme_opts & GNRC_LORAWAN_MLME_OPTS_LINK_CHECK_REQ) ?
        _fopts_mlme_link_check_req(mac, buf) : 0;
```

Reply:
this list is intended to be expanded if there are more fopts there. So I wrote it this way to make it easier to expand